src/plugins/visualize/public/application/components/visualize_editor_common.tsx

```diff
@@ -73,6 +85,9 @@ export const VisualizeEditorCommon = ({
         />
       )}
       {visInstance?.vis?.type?.stage === 'experimental' && <ExperimentalVisInfo />}
+      {/* Adds a deprecation warning for vislib xy axis charts */}
+      {/* Should be removed when this issue is closed https://github.com/elastic/kibana/issues/103209 */}
+      {isXYAxis(visInstance?.vis.type.name) && hasXYLegacyChartsEnabled && <DeprecationWarning />}
```

Comment:
You're going to show this deprecation warning to all users, but if possible we should only show this to users who have permissions to edit the advanced settings.

Reply:
@wylieconlon this is correct. Or maybe we could show this to everyone and if the user doesn't have the permissions to change the text a bit

So for example:
-  User has permissions
`You are using the legacy XY charts library which will be removed in the next minor release, 7.16, after being deprecated in 7.12. Switch to the new library in your advanced settings.`

- User doesn't have permissions
`You are using the legacy XY charts library which will be removed in the next minor release, 7.16, after being deprecated in 7.12. Contact your administrator to switch to the new library in your advanced settings.`

I don't want this notification to be lost for the users that don't have edit permissions. Wdyt?